packages/core/__tests__/utils/nuxt/proxyUtils.spec.ts

```diff
@@ -29,64 +31,84 @@ describe('[CORE - utils] _proxyUtils', () => {
   });
 
   it('it combines config with the current one', () => {
-    jest.spyOn(utils, 'getCookies').mockReturnValueOnce('');
-
-    expect(utils.getIntegrationConfig(
+    jest.spyOn(utils, 'getCookies').mockReturnValue('');
+    const integrationConfig = utils.getIntegrationConfig(
       {
         $config: {
           middlewareUrl: 'http://localhost.com'
         }
       } as any,
       { someGivenOption: 1 }
-    )).toEqual({
+    );
+
+    expect(integrationConfig).toEqual({
       axios: {
-        baseURL: 'http://localhost.com/api',
+        baseURL: 'http://localhost.com',
         headers: {}
       },
       someGivenOption: 1
     });
   });
 
   it('it combines config with the current one and adds a cookie', () => {
-    jest.spyOn(utils, 'getCookies').mockReturnValueOnce('xxx');
+    jest.spyOn(utils, 'getCookies').mockReturnValue('xxx');
 
-    expect(utils.getIntegrationConfig(
+    const integrationConfig = utils.getIntegrationConfig(
       {
         $config: {
           middlewareUrl: 'http://localhost.com'
         }
       } as any,
       {}
-    )).toEqual({
+    );
+
+    expect(integrationConfig).toEqual({
       axios: {
-        baseURL: 'http://localhost.com/api',
+        baseURL: 'http://localhost.com',
         headers: {
           cookie: 'xxx'
         }
       }
     });
   });
 
-  it('it combines config with the current one and adds a Host header', () => {
-    expect(utils.getIntegrationConfig(
+  /**
+   * baseURL configuration cases matrix
+   */
+  const urlSetupCases = [
+    { server: true, middlewareUrl: 'client/api', ssrMiddlewareUrl: 'server/api', expected: '/server/api' },
+    { server: false, middlewareUrl: 'client/api', ssrMiddlewareUrl: 'server/api', expected: '/client/api' },
+    { server: true, middlewareUrl: '/client/api', ssrMiddlewareUrl: '', expected: '/client/api' },
+    { server: false, middlewareUrl: 'https://client/api', ssrMiddlewareUrl: null, expected: 'https://client/api' },
+    { server: true, middlewareUrl: 'https://client/api', ssrMiddlewareUrl: 'https://server/api', expected: 'https://server/api' }
+  ];
+
+  const testMsg = '[baseUrl must be configured properly for] server: $server, middlewareUrl: $middlewareUrl, ssrMiddlewareUrl: $ssrMiddlewareUrl, expected: $expected';
+  test.each(urlSetupCases)(testMsg, ({ server, middlewareUrl, ssrMiddlewareUrl, expected }) => {
+    process.server = server;
+    const integrationConfig = utils.getIntegrationConfig(
```

Comment:
```suggestion
    
    const integrationConfig = utils.getIntegrationConfig(
```
We need an empty line between sections in AAA rule.